core/src/main/java/org/apache/carbondata/core/indexstore/blockletindex/BlockDataMap.java

```diff
@@ -507,7 +509,7 @@ private void addTaskSummaryRowToUnsafeMemoryStore(CarbonRowSchema[] taskSummaryS
     if (null != summaryRow) {
       summaryRow.setByteArray(fileName, SUMMARY_INDEX_FILE_NAME);
       summaryRow.setByteArray(segmentId, SUMMARY_SEGMENTID);
-      if (null != filePath) {
+      if (filePath.length > 0) {
```

Comment:
It better use:
`null != filePath && filePath.length > 0`

Reply:
The filePath is can't be null, so we uncheck it here.